src/python/pants/backend/python/dependency_inference/parse_python_dependencies.py

```diff
@@ -120,6 +120,7 @@ async def parse_python_dependencies(
             if (
                 python_infer_subsystem.string_imports
                 and not slash_count
+                and not string.endswith(".")
                 and string.count(".") >= python_infer_subsystem.string_imports_min_dots
```

Comment:
```suggestion
                python_infer_subsystem.string_imports
                and string.count(".") >= python_infer_subsystem.string_imports_min_dots
                and all(part.isidentifier() for part in string.split("."))
```

Let's just switch to having Python tell us what's valid Python 😅 

I'm not sure how/if we should handle entry string. Let's assume we'd never get an empty string because that's the logical thing to do.

```python
>>> bool(all(s.isidentifier() for s in "".split()))
True
>>> bool(all(s.isidentifier() for s in "a.".split()))
False
>>> bool(all(s.isidentifier() for s in "a".split()))
True
>>> bool(all(s.isidentifier() for s in "a/b".split()))
False
```

Reply:
I didn't know about the `isidentifier` method. Nice!

You've got a typo in your test:
You need `split(".")` instead of `split()`

```
>>> all(s.isidentifier() for s in "".split("."))
False
>>> all(s.isidentifier() for s in ".".split("."))
False
>>> all(s.isidentifier() for s in "a.".split("."))
False
>>> all(s.isidentifier() for s in "a".split("."))
True
>>> all(s.isidentifier() for s in "a/b".split("."))
False
```

Reply:
K. I just pushed a commit that adopts your suggestion and postpones the rust changes for a future PR.

Reply:
You're the best ♥️